spot-oa/oa/dns/ipynb_templates/Threat_Investigation_master.ipynb

```diff
@@ -160,24 +160,17 @@
     "                if row[1] != '' : c_dns.append(row[1])\n",
     "            \n",
     "    with open(sconnect, 'r') as f:\n",
-    "        reader = csv.reader(f, delimiter=',')\n",
-    "        reader.next()\n",
+    "        reader = csv.DictReader(f, delimiter=',')\n",
     "        for row in reader:\n",
-    "            # frame_time, frame_len, ip_dst, dns_qry_name, dns_qry_class,  dns_qry_type, dns_qry_rcode, domain, subdomain, \n",
-    "            #      0    ,      1   ,    2  ,       3     ,          4   ,         5    ,        6     ,   7   ,    8     ,\n",
-    "            #subdomain_length, num_periods, subdomain_entropy, top_domain, word, score, query_rep,  hh, \n",
-    "            #      9          ,      10    ,     11          ,    12     ,  13 ,  14  ,   15     ,  16, \n",
-    "            #ip_sev, dns_sev, dns_qry_class_name, dns_qry_type_name, dns_qry_rcode_name, network_context, unix_tstamp\n",
-    "            # 17   ,   18   ,           19      ,      20          ,         21        ,      22        ,     23\n",
-    "            if row[2] not in ips_query and row[2] not in c_ips and row[17] == '1': \n",
-    "                    ips_query[row[2]]='i'\n",
-    "            if row[3] not in ips_query and row[3] not in c_dns and row[18] == '1':\n",
-    "                    ips_query[row[3]]='q' \n",
+    "            if row['ip_dst'] not in ips_query and row['ip_dst'] not in c_ips and row['ip_sev'] == '1': \n",
+    "                    ips_query[row['ip_dst']]='i'\n",
+    "            if row['dns_qry_name'] not in ips_query and row['dns_qry_name'] not in c_dns and row['dns_sev'] == '1':\n",
+    "                    ips_query[row['dns_qry_name']]='q' \n",
     "            \n",
-    "            if row[2] not in ip_sev: \n",
-    "                ip_sev[row[2]] = row[14]\n",
-    "            if row[3] not in dns_sev: \n",
-    "                dns_sev[row[3]] =row[14]\n",
+    "            if row['ip_dst'] not in ip_sev: \n",
+    "                ip_sev[row['ip_dst']] = row['score']\n",
+    "            if row['dns_qry_name'] not in dns_sev: \n",
+    "                dns_sev[row['dns_qry_name']] =row['score']\n",
```

Comment:
I like them too, but I like them even more when there's a named constant (Eg. IP_DST = 'ip_dst') so that typos in the column names are found by static syntax checking and not just at run time